plugins/datastreamer_plugin/datastreamer/FastDdsDataStreamer.cpp

```diff
@@ -0,0 +1,233 @@
+// Copyright 2022 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// This file is part of eProsima Fast DDS Monitor.
+//
+// eProsima Fast DDS Monitor is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// eProsima Fast DDS Monitor is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with eProsima Fast DDS Monitor. If not, see <https://www.gnu.org/licenses/>.
+
+/**
+ * @file FastDdsDataStreamer.cpp
+ */
+
+#include "FastDdsDataStreamer.hpp"
+#include "ui/topic_selection_dialog/dialogselecttopics.h"
+#include "utils/utils.hpp"
+
+namespace eprosima {
+namespace plotjuggler {
+namespace datastreamer {
+
+const char* FastDdsDataStreamer::PLUGIN_NAME_ = "Fast DDS DataStreamer PlotJuggler Plugin";
+
+FastDdsDataStreamer::FastDdsDataStreamer()
+    : running_(false)
+    , fastdds_handler_(std::make_unique<fastdds::Handler>(this))
+    , configuration_(std::make_shared<ui::Configuration>())
+    , select_topics_dialog_(std::make_unique<ui::DialogSelectTopics>(
+                configuration_,
+                fastdds_handler_->get_topic_data_base(),
+                this))
```

Comment:
Here I have my doubts. I agree `Handler` and `DialogSelectTopics` must be inner objects (the reason they are not is the hurry while implementing this). But `Configuration` is an inner object (which owner is `FastDdsDataStreamer`, but it will be used in other objects (DialogSelectTopics), so it should send the DialogSelectTopics the pointer (&configuration_), the reference (Configuration& configuration) or use a `shared_ptr`?

Reply:
* Use shared/unique pointers are needed when lifetime semantics are needed. If no lifetime semantics are required (the owner of a resource is clear and its lifetime is clear. Mere resource use is not a sufficient condition to pass a smart pointer. So use smart pointers as a last resort, not by default, according to the multiple reasons said in the CG.
* As core guidelines say, pass a raw ptr or reference to users, when users are not meant to control the lifetime (almost always).
* Pass and set a reference if the object is always defined (references cannot be null-initialized), so we provide additional compile-time check - compiler will complain if a reference is not initialized.
* Pass and set a pointer if the resource is optionally initialized (pointers can be null-initialized - nullptr), so the interface is flexible enough to accommodate to non-initialized cases